v2/devdoc/log_context_property_type_wchar_t_ptr_requirements.md

```diff
@@ -0,0 +1,95 @@
+
+## Exposed API
+
+```
+int LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(void* dst_value, size_t count, const wchar_t* format, ...);
+int LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)(const wchar_t* format, ...);
+
+extern const LOG_CONTEXT_PROPERTY_TYPE_IF LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr);
+
+```
+
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_TO_STRING)(const void* property_value, char* buffer, size_t buffer_length);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` produces the string representation of a `wchar_t_ptr`.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_001: [** If `property_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` shall fail and return a negative value.  **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_002: [** If `buffer` is `NULL` and `buffer_length` is greater than 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` shall fail and return a negative value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_003: [** If `buffer` is `NULL` and `buffer_length` is 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` shall return the length of the `wchar_t` string pointed to by `property_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_004: [** Otherwise, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` shall copy the wchar_t string pointed to by `property_value` to `buffer` by using `snprintf` with `buffer`, `buffer_length` and format string `%ls` and pass in the values list the `const wchar_t*` value pointed to be `property_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_005: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` shall succeed and return the result of `snprintf`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_006: [** If any error is encountered (truncation is not an error), `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` shall fail and return a negative value. **]**
+
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).copy
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_COPY)(void* dst_value, const void* src_value);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).copy` copies the `wchar_t_ptr` value from the address pointed by `src_value` to `dst_value`.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_007: [** If `src_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).copy` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_008: [** If `dst_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).copy` shall fail and return a non-zero value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_009: [** `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).copy` shall copy the entire string (including the null terminator) from `src_value` to `dst_value`. **]**
```

Comment:
>copy [](http://example.com/codeflow?start=116&length=4)

Just wondering at this moment what are the guarantees that `dst_value` has enough memory to accomodate `src_value`.

Reply:
no guarantees, change to use wcscpy_s here and use strcpy_s in ascii_char_ptr_log_context_property_type_copy

Reply:
strcpy_s/wcscpy_s not  supported on linux, don't know how to make it safe

Reply:
well - if there are "no guarantees" then it might overflow and overwrite some other parts of memory. 

Who calls this function... must have prepared "dst_value" somehow. So if there are any guarantees, they come from the preparer of `dst_value`. 

Can you please trace from where is dst_value coming and see if its memory size matches src_value?

Reply:
done by convention in log_context.h